proto/ligato/generic/model.proto

```diff
@@ -1,51 +1,43 @@
-syntax = "proto3";
+syntax = "proto2";
 
 package ligato.generic;
 
 option go_package = "go.ligato.io/vpp-agent/v2/proto/ligato/generic";
 
+import "google/protobuf/descriptor.proto";
+
 // ModelSpec defines a model specification to identify a model.
 message ModelSpec {
 
     // Module describes grouping for the model.
-    string module = 1;
+    optional string module = 1;
+
     // Version describes version of the model schema.
-    string version = 2;
+    optional string version = 2;
+
     // Type describes name of type described by this model.
-    string type = 3;
+    optional string type = 3;
+
     // Class describes purpose for the model.
-    string class = 4;
+    optional string class = 4;
 }
 
-// ModelDescriptor defines model descriptor to describe a model.
-message ModelDescriptor {
+extend google.protobuf.MessageOptions {
+    optional ModelSpec model = 50222;
+}
+
+// ModelDetail represents info about model details.
+message ModelDetail {
 
     // Spec is a specificaiton the model was registered with.
-    ModelSpec spec = 1;
+    required ModelSpec spec = 1;
+
     // ProtoName is a name of protobuf message representing the model.
-    string proto_name = 2;
+    optional string proto_name = 2;
 
     message Option {
-        string key = 1;
+        required string key = 1;
```

Comment:
I see the meaning of the optional/required fields here, but wouldn't it be a better approach to solve it programatically? Using proto2 is fine, but switching just because of this is unnecessary.

Reply:
This was not the reason behind using `proto2`. It was required to allow extension, but I moved it to a separate file to avoid defining this in `proto2`.